### PR TITLE
fix podman, update some stuff, remove redundant stuff

### LINKED ISSUE #2581: Make docker setup work with podman

### PR DESCRIPTION
**Pull request by sitarane**
#### Proposed Changes

* Add volume instruction that makes it work with rootless containers
* Remove redundancy in environment variable declaration
* Remove unnecessary fluff from docker-compose.yml

#### Testing Instructions

* Install docker if you don't have it yet: https://docs.docker.com/engine/install/
* Do the config/env/local.js thing described here: https://team.trustroots.org/Install-Docker.html
* docker-compose up
* should work (localhost:3000)

Fixes #2581

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,8 +1,4 @@
-version: '3'
-
-networks:
-  trustroots-network:
-    driver: bridge
+version: '3.8'
```
Comment (sitarane) [inline docker-compose.yml:1]: Never hurts to be on the latest version

File: docker-compose.yml [hunk 2]
```diff
@@ -11,22 +7,16 @@ volumes:
 
 services:
   trustroots:
-    image: node:14
```
Comment (sitarane) [inline docker-compose.yml:14]: We build the image. It is defined in the Dockerfile.

File: docker-compose.yml [hunk 3]
```diff
@@ -11,22 +7,16 @@ volumes:
...
-    container_name: trustroots
```
Comment (sitarane) [inline docker-compose.yml:15]: containers get named by the service name. Redundant.

File: docker-compose.yml [hunk 4]
```diff
@@ -11,22 +7,16 @@ volumes:
...
-      context: .
-      dockerfile: Dockerfile
+    build: .
     # Mount the app dir in the container so our changes
     #  to the app code are also changed in the container
     #
     # Mount node_modules separately as a data volume, so
     #  that they won't get overwritten by app directory
     #  Read more: http://stackoverflow.com/a/32785014/1984644
-    networks:
```
Comment (sitarane) [inline docker-compose.yml:26]: A network is automatically created for linked services.

File: docker-compose.yml [hunk 5]
```diff
@@ -11,22 +7,16 @@ volumes:
...
-      - trustroots-network
     volumes:
-      - ./:/trustroots
+      - .:/trustroots:Z # Z for rootless docker users
```
Comment (sitarane) [inline docker-compose.yml:19]: Just this fixes the issue. Everything else is negociable :)

File: docker-compose.yml [hunk 6]
```diff
@@ -39,15 +29,13 @@ services:
       - '35729:35729'
     environment:
       - DB_1_PORT_27017_TCP_ADDR=mongodb
+      - NODE_ENV=development
 
   ## Database container
   mongodb:
     image: mongo:4.4
-    container_name: db_1
```
Comment (sitarane) [inline docker-compose.yml:46]: I removed this as it doesn't seem to be used anywhere.

File: docker-compose.yml [hunk 7]
```diff
@@ -39,15 +29,13 @@ services:
...
     volumes:
       - dbdata:/data/db
     restart: unless-stopped
-    ports:
```
Comment (sitarane) [inline docker-compose.yml:50]: This isn't necessary for the trustroots service. It would be necessary for the *host* to access the db.

**Comment by sitarane** [comment]
I borrowed a laptop, installed docker, and got the environment to run just fine with this PR. But it'd be nice if someone else tested it.

**Comment by mariha** [comment]
I tried it and I am getting

```
trustroots_1  | [webpack-cli] Failed to load '/trustroots/config/webpack/webpack.config.js' config
trustroots_1  | [webpack-cli] Error: Cannot find module 'eslint-webpack-plugin'
trustroots_1  | Require stack:
trustroots_1  | - /trustroots/config/webpack/webpack.config.js
trustroots_1  | - /trustroots/node_modules/webpack-cli/lib/webpack-cli.js
trustroots_1  | - /trustroots/node_modules/webpack-cli/lib/bootstrap.js
trustroots_1  | - /trustroots/node_modules/webpack-cli/bin/cli.js
trustroots_1  | - /trustroots/node_modules/webpack/bin/webpack.js
trustroots_1  |     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
trustroots_1  |     at Function.Module._load (internal/modules/cjs/loader.js:746:27)
trustroots_1  |     at Module.require (internal/modules/cjs/loader.js:974:19)
trustroots_1  |     at require (/trustroots/node_modules/webpack-cli/node_modules/v8-compile-cache/v8-compile-cache.js:159:20)
trustroots_1  |     at Object.<anonymous> (/trustroots/config/webpack/webpack.config.js:7:22)
trustroots_1  |     at Module._compile (/trustroots/node_modules/webpack-cli/node_modules/v8-compile-cache/v8-compile-cache.js:192:30)
trustroots_1  |     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
trustroots_1  |     at Module.load (internal/modules/cjs/loader.js:950:32)
trustroots_1  |     at Function.Module._load (internal/modules/cjs/loader.js:790:14)
trustroots_1  |     at Module.require (internal/modules/cjs/loader.js:974:19) {
trustroots_1  |   code: 'MODULE_NOT_FOUND',
trustroots_1  |   requireStack: [
trustroots_1  |     '/trustroots/config/webpack/webpack.config.js',
trustroots_1  |     '/trustroots/node_modules/webpack-cli/lib/webpack-cli.js',
trustroots_1  |     '/trustroots/node_modules/webpack-cli/lib/bootstrap.js',
trustroots_1  |     '/trustroots/node_modules/webpack-cli/bin/cli.js',
trustroots_1  |     '/trustroots/node_modules/webpack/bin/webpack.js'
trustroots_1  |   ]
trustroots_1  | }
trustroots_1  | npm ERR! code ELIFECYCLE
trustroots_1  | npm ERR! errno 2
trustroots_1  | npm ERR! trustroots@0.3.3 webpack:server: `webpack serve --hot --config config/webpack/webpack.config.js`
trustroots_1  | npm ERR! Exit status 2
trustroots_1  | npm ERR! 
trustroots_1  | npm ERR! Failed at the trustroots@0.3.3 webpack:server script.
trustroots_1  | npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
trustroots_1  | 
trustroots_1  | npm ERR! A complete log of this run can be found in:
trustroots_1  | npm ERR!     /root/.npm/_logs/2022-08-18T04_43_34_932Z-debug.log
```

Locally on my machine (without docker) I build and run TR code with node 16. In Dockerfile we use node 14. I tried both and none of them works for me.

**Comment by sitarane** [comment]
@mariha : thanks for testing. I'm assuming you tried setting up TR the traditional way. I tried as well, and it's working.

My setup: Node v16.17.0 / Mongo 4.4.4
I followed the instructions there: https://team.trustroots.org/Install.html

If that's not what you did, could you describe what you did to get to that point?

**Pull request closed without merging**